src/PerformanceMonitor/PerformanceMonitor.BlockProperties.cs

```diff
@@ -0,0 +1,242 @@
+
+            if (block is null)
+                throw new RpcException(-100, "Block not found");
+
+            return block.GetTime();
+        }
+
+        /// <summary>
+        /// Get a block identified by its index or hash
+        /// </summary>
+        /// <param name="blockIndexOrHash">
+        /// Index or hash of the desired block
+        /// </param>
+        /// <returns>
+        /// Returns the block if the index or hash exists; otherwise,
+        /// returns null
+        /// </returns>
+        private Block GetBlock(string blockIndexOrHash)
+        {
+            Block block = null;
+
+            if (UInt256.TryParse(blockIndexOrHash, out var blockHash))
+            {
+                block = Blockchain.Singleton.GetBlock(blockHash);
+            }
+            else if (uint.TryParse(blockIndexOrHash, out var blockIndex))
+            {
+                block = Blockchain.Singleton.GetBlock(blockIndex);
+            }
+
+            return block;
+        }
+
+        /// <summary>
+        /// Process "block avgtime" command
+        /// Prints the average time in seconds the latest blocks are active
+        /// </summary>
+        [ConsoleCommand("block avgtime", Category = "Block Commands", Description = "Show the average time in seconds the latest blocks are active.")]
+        private void OnBlockAverageTimeCommand(uint blockCount = 1000)
+        {
+            uint desiredCount = blockCount;
+
+            if (desiredCount < 1)
+            {
+                Console.WriteLine("Minimum 1 block");
+                return;
+            }
+
+            if (desiredCount > 10000)
+            {
+                Console.WriteLine("Maximum 10000 blocks");
+                return;
+            }
+
+            using (var snapshot = Blockchain.Singleton.GetSnapshot())
+            {
+                var averageInSeconds = snapshot.GetAverageTimePerBlock(desiredCount) / 1000;
+                Console.WriteLine(averageInSeconds.ToString("Average time/block: 0.00 seconds"));
+            }
+        }
+
+        /// <summary>
+        /// Gets the average time the latest blocks are active
+        /// </summary>
+        /// <returns>
+        /// Returns the average time per block in milliseconds
+        /// </returns>
+        [RpcMethod]
+        public JObject GetBlockAvgTime(JArray _params)
+        {
+            if (_params.Count > 1)
+            {
+                throw new RpcException(-32602, "Invalid params");
+            }
+            uint desiredCount = 1000;
+            if (_params.Count > 0)
+            {
+                if (!uint.TryParse(_params[0].AsString(), out desiredCount))
+                {
+                    throw new RpcException(-32602, "Invalid params");
+                }
+
+                if (desiredCount < 1)
+                {
+                    throw new RpcException(-100, "Minimum 1 block");
+                }
+
+                if (desiredCount > 10000)
```

Comment:
How much time spend with this limit?

Reply:
Using RPC, I got 889 ms in total time, with this method using 168 ms in the RPC Server
![image](https://user-images.githubusercontent.com/19419485/87329981-440ae480-c50e-11ea-8990-46c7b009d4d6.png)
![image](https://user-images.githubusercontent.com/19419485/87330022-54bb5a80-c50e-11ea-9d3c-22b85f75a2eb.png)
Do you think we should decrease this maximum as well?